[CI] SnapshotDisruptionIT. testDisruptionOnSnapshotInitialization fails unreproducibly

See https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/837/console

Ran this with hundred iterations on Linux and once on osx, does not reproduce. Reproduction line is

```
 ./gradlew :server:integTest \
  -Dtests.seed=642A19B3B0D37154 \
  -Dtests.class=org.elasticsearch.discovery.SnapshotDisruptionIT \
  -Dtests.method="testDisruptionOnSnapshotInitialization" \
  -Dtests.security.manager=true \
  -Dtests.locale=hr-HR \
  -Dtests.timezone=Europe/Tirane \
  -Dcompiler.java=11 \
  -Druntime.java=8
```

I saw another test failure in the last days. Log excerpt

```
  1> [2018-12-18T14:58:52,986][INFO ][o.e.d.SnapshotDisruptionIT] [testDisruptionOnSnapshotInitialization] after test
FAILURE 4.73s J1 | SnapshotDisruptionIT.testDisruptionOnSnapshotInitialization <<< FAILURES!
   > Throwable #1: java.lang.AssertionError:
   > Expected: an instance of org.elasticsearch.discovery.MasterNotDiscoveredException
   >      but: <RemoteTransportException[[node_tm1][127.0.0.1:41899][cluster:admin/snapshot/create]]; nested: ConcurrentSnaps
hotExecutionException[[test-repo:test-snap-2] cannot snapshot while a snapshot deletion is in-progress];> is a org.elasticsea
rch.transport.RemoteTransportException
   >    at __randomizedtesting.SeedInfo.seed([642A19B3B0D37154:4EF53F6396CB828B]:0)
   >    at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   >    at org.elasticsearch.discovery.SnapshotDisruptionIT.testDisruptionOnSnapshotInitialization(SnapshotDisruptionIT.java:
164)
   >    at java.lang.Thread.run(Thread.java:748)

```

Direct log https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/837/consoleText

Failure in this build https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/1189/console

```
Publication{term=2, version=38}
  1> [2019-01-10T13:22:04,999][INFO ][o.e.d.SnapshotDisruptionIT] [testDisruptionOnSnapshotInitialization] --> done
  1> [2019-01-10T13:22:05,007][WARN ][o.e.s.SnapshotsService   ] [node_tm0] [test-repo][test-snap-2] failed to create snapshot
  1> org.elasticsearch.snapshots.ConcurrentSnapshotExecutionException: [test-repo:test-snap-2] cannot snapshot while a snapshot deletion is in-progress
  1> 	at org.elasticsearch.snapshots.SnapshotsService$1.execute(SnapshotsService.java:255) ~[main/:?]
  1> 	at org.elasticsearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:47) ~[main/:?]
  1> 	at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:686) ~[main/:?]
  1> 	at org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:311) ~[main/:?]
  1> 	at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:211) [main/:?]
  1> 	at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:143) [main/:?]
  1> 	at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:150) [main/:?]
  1> 	at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:188) [main/:?]
  1> 	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:660) [main/:?]
  1> 	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:244) [main/:?]
  1> 	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:207) [main/:?]
  1> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_192]
  1> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_192]
  1> 	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_192]
```

```
FAILURE 71.0s J1 | SnapshotDisruptionIT.testDisruptionOnSnapshotInitialization <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: SnapshotMissingException[[test-repo:test-snap-2] is missing]
   > 	at __randomizedtesting.SeedInfo.seed([DD7787B0D526D4B7:F7A8A160F33E2768]:0)
   > 	at org.elasticsearch.discovery.SnapshotDisruptionIT.lambda$testDisruptionOnSnapshotInitialization$2(SnapshotDisruptionIT.java:185)
   > 	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:847)
   > 	at org.elasticsearch.discovery.SnapshotDisruptionIT.testDisruptionOnSnapshotInitialization(SnapshotDisruptionIT.java:181)
   > 	at java.lang.Thread.run(Thread.java:748)
   > 	Suppressed: java.lang.AssertionError: SnapshotMissingException[[test-repo:test-snap-2] is missing]
   > 		at org.elasticsearch.discovery.SnapshotDisruptionIT.lambda$testDisruptionOnSnapshotInitialization$2(SnapshotDisruptionIT.java:185)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
   > 		... 38 more
   > 	Caused by: SnapshotMissingException[[test-repo:test-snap-2] is missing]
   > 		at org.elasticsearch.action.admin.cluster.snapshots.get.TransportGetSnapshotsAction.masterOperation(TransportGetSnapshotsAction.java:115)
   > 		at org.elasticsearch.action.admin.cluster.snapshots.get.TransportGetSnapshotsAction.masterOperation(TransportGetSnapshotsAction.java:54)
   > 		at org.elasticsearch.action.support.master.TransportMasterNodeAction.masterOperation(TransportMasterNodeAction.java:127)
   > 		at org.elasticsearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction$2.doRun(TransportMasterNodeAction.java:208)
   > 		at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:759)
   > 		at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
   > 		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
   > 		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
   > 		... 1 more
   > 	Suppressed: java.lang.AssertionError: SnapshotMissingException[[test-repo:test-snap-2] is missing]
   > 		at org.elasticsearch.discovery.SnapshotDisruptionIT.lambda$testDisruptionOnSnapshotInitialization$2(SnapshotDisruptionIT.java:185)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
   > 		... 38 more
   > 	Caused by: SnapshotMissingException[[test-repo:test-snap-2] is missing]
```

--- a/server/src/test/java/org/elasticsearch/discovery/SnapshotDisruptionIT.java
+++ b/server/src/test/java/org/elasticsearch/discovery/SnapshotDisruptionIT.java
@@ -133,21 +133,7 @@ public class SnapshotDisruptionIT extends ESIntegTestCase {
         logger.info("--> waiting for disruption to start");
         assertTrue(disruptionStarted.await(1, TimeUnit.MINUTES));
 
-        logger.info("--> wait until the snapshot is done");
-        assertBusy(() -> {
-            ClusterState state = dataNodeClient().admin().cluster().prepareState().get().getState();
-            SnapshotsInProgress snapshots = state.custom(SnapshotsInProgress.TYPE);
-            SnapshotDeletionsInProgress snapshotDeletionsInProgress = state.custom(SnapshotDeletionsInProgress.TYPE);
-            if (snapshots != null && snapshots.entries().size() > 0) {
-                logger.info("Current snapshot state [{}]", snapshots.entries().get(0).state());
-                fail("Snapshot is still running");
-            } else if (snapshotDeletionsInProgress != null && snapshotDeletionsInProgress.hasDeletionsInProgress()) {
-                logger.info("Current snapshot deletion state [{}]", snapshotDeletionsInProgress);
-                fail("Snapshot deletion is still running");
-            } else {
-                logger.info("Snapshot is no longer in the cluster state");
-            }
-        }, 1, TimeUnit.MINUTES);
+        assertAllSnapshotsCompleted();
 
         logger.info("--> verify that snapshot was successful or no longer exist");
         assertBusy(() -> {
@@ -177,14 +163,25 @@ public class SnapshotDisruptionIT extends ESIntegTestCase {
             }
         }
 
-        logger.info("--> verify that snapshot eventually will be created due to retries");
+        assertAllSnapshotsCompleted();
+    }
+
+    private void assertAllSnapshotsCompleted() throws Exception {
+        logger.info("--> wait until the snapshot is done");
         assertBusy(() -> {
-            try {
-                assertSnapshotExists("test-repo", "test-snap-2");
-            } catch (SnapshotMissingException ex) {
-                throw new AssertionError(ex);
+            ClusterState state = dataNodeClient().admin().cluster().prepareState().get().getState();
+            SnapshotsInProgress snapshots = state.custom(SnapshotsInProgress.TYPE);
+            SnapshotDeletionsInProgress snapshotDeletionsInProgress = state.custom(SnapshotDeletionsInProgress.TYPE);
+            if (snapshots != null && snapshots.entries().isEmpty() == false) {
+                logger.info("Current snapshot state [{}]", snapshots.entries().get(0).state());
+                fail("Snapshot is still running");
+            } else if (snapshotDeletionsInProgress != null && snapshotDeletionsInProgress.hasDeletionsInProgress()) {
+                logger.info("Current snapshot deletion state [{}]", snapshotDeletionsInProgress);
+                fail("Snapshot deletion is still running");
+            } else {
+                logger.info("Snapshot is no longer in the cluster state");
             }
-        }, 1, TimeUnit.MINUTES);
+        }, 1L, TimeUnit.MINUTES);
     }
 
     private void assertSnapshotExists(String repository, String snapshot) {

